common/src/main/java/org/opensearch/ml/common/connector/HttpConnector.java

```diff
@@ -324,6 +325,29 @@ public  <T> T createPayload(String action, Map<String, String> parameters) {
         return (T) parameters.get("http_body");
     }
 
+    private Map<String,String> formatArrayParameters(Map<String, String> parameters) {
+        Map<String,String> newParameters =  new HashMap<>();
+        for (Map.Entry<String, String> entry : parameters.entrySet()) {
+            String key = entry.getKey();
+            String value = entry.getValue();
+            String escapedValue = escapeJsonArrayIfNeeded(value);
+            newParameters.put(key,escapedValue);
+        }
+        return newParameters;
+    }
+    protected String escapeJsonArrayIfNeeded(String value) {
```

Comment:
Can you think further if this can extend to support an array of array?

Reply:
I try added a UT for array of array, that's a good point